chain/ethereum/src/network.rs

```diff
@@ -48,64 +72,129 @@ impl EthereumNetworkAdapter {
 
 #[derive(Debug, Clone)]
 pub struct EthereumNetworkAdapters {
-    pub adapters: Vec<EthereumNetworkAdapter>,
+    chain_id: ChainId,
+    manager: ProviderManager<EthereumNetworkAdapter>,
     call_only_adapters: Vec<EthereumNetworkAdapter>,
     // Percentage of request that should be used to retest errored adapters.
     retest_percent: f64,
 }
 
-impl Default for EthereumNetworkAdapters {
-    fn default() -> Self {
-        Self::new(None)
-    }
-}
-
 impl EthereumNetworkAdapters {
-    pub fn new(retest_percent: Option<f64>) -> Self {
+    pub fn empty() -> Self {
```

Comment:
Had to revert to renaming because of the build in release mode step on CI that builds the test crate for some reason 
![image](https://github.com/graphprotocol/graph-node/assets/1640325/f62b37f5-59bc-4004-92eb-bc0dec7c93e1)